### PR TITLE
Pip installable wheel

### PR DESCRIPTION
**Pull request by gsfr**
With @kofalt's help, I have gotten CI to build an installable Python wheel for the SDK.

Currently CI runs only on Linux, so we can only build for Linux.

The artifact is temporarily hosted on Circle.

Try this on a Linux host:
`pip install https://223-65499703-gh.circle-artifacts.com/0/bridge/python/flywheel-0.0.1-py2-none-linux_x86_64.whl`

Todos:
- [ ] Auto-build wheel for OS X
- [ ] Host wheels somewhere (pypi or GitHub)
- [ ] Define and implement wheel update strategy (e.g., only for tags)
- [ ] Document in bridge readme supported workflow for local build and test

### CODE REVIEW

**Comment by kofalt** [comment]
Personally I would be fine if OSX is left for #26, but if someone gets it working that would be great.
I'd like the pip command to be moved to a `bridge/dist.sh` and invoked from the config.yml.

It would also be nice if we can figure out a creative way to get the apt commands out of there, but Gunnar and I have already discussed some of the challenges with that :/

**Comment by gsfr** [comment]
@kofalt I have reduced the `apt-get` to what I believe is the minimum required to build the wheels. I've also added python3 support. I'm stopping here. Feel free to reorganize as desired.

@ryansanford and I have decided to go ahead with maintaining our own utility images. We seem to have reached critical mass. I hope to get a GitHub repo started today, basically just to hold Dockerfiles, linked to Dockerhub auto builds.

**Comment by kofalt** [comment]
Request that you condense [this](https://github.com/flywheel-io/sdk/blob/9039bba3b9e6378fba88669d9077265fab7f179b/.circleci/config.yml#L34-L71) into an image! We'll be using that in multiple circle repos.

**Comment by kofalt** [comment]
Head's up all: I rebased onto master to grab #34 for @jenreiter :+1:

**Comment by gsfr** [comment]
@ryansanford Latest commit works with `docker run ubuntu` and `apt-get install python3-pip python-pip` under both python 2 and 3.

**Comment by kofalt** [comment]
Rebasing again for Jen's benefit

**Comment by kofalt** [comment]
@gsfr, added a quick commit keeping the versions in sync. If you have a reason you'd like the pip version to be distinct, let me know. I'm assuming we don't?

**Comment by kofalt** [comment]
Could setup.py run some code and determine that value? I could give you a script that outputs the number. Presumably that code would only run when *building* the wheel, not *using* it?

**Comment by gsfr** [comment]
That might work, but I'll have to look into the details of setup.py execution. Let's keep things sync'ed by hand until I get to that.

File: .circleci/config.yml [hunk 1]
```diff
@@ -127,15 +127,31 @@ jobs:
...
-                command: python test-drive.py
+                name: Build and Test Wheel
+                command: |
+                  python2 setup.py bdist_wheel --plat-name linux_x86_64 --dist-dir wheelhouse
+                  python3 setup.py bdist_wheel --plat-name linux_x86_64 --dist-dir wheelhouse
+                  cd wheelhouse
+                  pip install "$(find -name *py2*.whl)"
+                  python ../test-drive.py
+                  pip3 install "$(find -name *py3*.whl)"
+                  python3 ../test-drive.py
```
Comment (kofalt) [inline .circleci/config.yml]: The bdist-commands should move to a `dist.sh` script in the bridge folder

File: upload-dist.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+set -euo pipefail
+unset CDPATH; cd "$( dirname "${BASH_SOURCE[0]}" )"; cd "$(pwd -P)"
```
Comment (kofalt) [inline upload-dist.sh]: Move this file to the bridge folder and adjust the `cd` command

**Comment by ryansanford** [comment]
@lmperry @jenreiter @gsfr  Looking for some input related to OSX pip install.

Essentially, we are requiring a version of the `six` library incompatible with that included with the OSX system Python libraries. I don't think we want to include instructions that will bypass this protection of the OSX managed python environment. There's a couple options in my mind.

1. Have instructions demonstrating use of venv with OSX system python
2. Have install instructions demonstrating use of homebrew/macports/python.org python
3. Some combination of above?
4. Other?

Not sure what is common or expected among scientific community for using python on osx

```
Ryans-New-MacBook-Pro:python ryansanford$ pip install wheelhouse/flywheel-0.2.0-py2-none-macosx_10_12_intel.whl
Processing ./wheelhouse/flywheel-0.2.0-py2-none-macosx_10_12_intel.whl
Collecting six>=1.10.0 (from flywheel==0.2.0)
  Downloading six-1.11.0-py2.py3-none-any.whl
Installing collected packages: six, flywheel
  Found existing installation: six 1.4.1
    DEPRECATION: Uninstalling a distutils installed project (six) has been deprecated and will be removed in a future version. This is due to the fact that uninstalling a distutils project will only partially uninstall the project.
    Uninstalling six-1.4.1:
Exception:
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_set.py", line 778, in install
    requirement.uninstall(auto_confirm=True)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_install.py", line 754, in uninstall
    paths_to_remove.remove(auto_confirm)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/req/req_uninstall.py", line 115, in remove
    renames(path, new_path)
  File "/Library/Python/2.7/site-packages/pip-9.0.1-py2.7.egg/pip/utils/__init__.py", line 267, in renames
    shutil.move(old, new)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 302, in move
    copy2(src, real_dst)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 131, in copy2
    copystat(src, dst)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 103, in copystat
    os.chflags(dst, st.st_flags)
OSError: [Errno 1] Operation not permitted: '/var/folders/2v/214htyss3256vb_h67xvqdw80000gp/T/pip-Orr4u6-uninstall/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/six-1.4.1-py2.7.egg-info'

```

**Comment by kofalt** [comment]
I don't think the update is "incompatible", you're just getting a mac problem that has been around for a few years: pypa/pip#3165. The summary sounds like another "Apple installs & locks old versions of important packages" problem. 

My suggestion would be to have OSX users run `sudo pip install --upgrade --ignore-installed six` before installing our library. This sounds like another "unbreak osx" step that mac users just have to deal with, same as `brew install bash`.

**Comment by ryansanford** [comment]
@kofalt I'm not sure what the exact dependency on `six` is, but if we're ok with ignoring the upgrade of it from `1.4.1` on my osx, presumably we should change our setup.py to not require a more recent version?

Maybe I should have started with the question, "How far back do we want to support OSX managed python installs"

**Comment by kofalt** [comment]
The ignore-existing flag only means "ignore the existing version if one exists". I believe, based on reading that ticket, that the newer `six` will end up in an actually-modifyable directory, and get imported first by future use (because python-path). There's a few comments there indicating that. I could be wrong?

So to directly answer your question: I don't think it applies, because userspace-python will override it with that command.

**Comment by ryansanford** [comment]
@kofalt I may have assumed too much. I'll give it a spin to confirm what that parameter does/doesn't do

**Comment by ryansanford** [comment]
@kofalt On OSX 10.11.6 the command `sudo pip install --upgrade --ignore-installed six` completed without error, making it look like the latest `six` package was installed, however, I still get the same error installing our wheel about operation not permitted. `pip freeze` makes it look like the latest `six` package is not in effect.

This is done within a VM not supporting copy/paste. I'll try setting up ssh to record here exact commands and result.

**Comment by ryansanford** [comment]
Update on my last comment about `--ignore-installed`. Despite pip printing that the newer `six` is installed, is not actually installed anywhere.

The default OSX version of `six` is still what ends up being resolved. Need to do some testing to confirm what version of `six` is actually required.

**Comment by kofalt** [comment]
Rebased.

It definitely sounds like a foundational platform issue outside our control, so as long as we're reasonably confident we can get OSX users up & running in some fashion I would be a fan of moving this towards a merge.

**Comment by kofalt** [comment]
8fe1be628064db3660320dc8abc4011b

I am absconding with this branch.

* Linux wheels can be downloaded from Circle
* Mac wheels can be built manually
* Github releases are disabled for now

Future PRs welcome.

**Pull request merged**